howto/configure-mtls/README.md

```diff
@@ -0,0 +1,198 @@
+
+## Sentry Configuration
+
+mTLS settings reside in a Dapr configuration file.
+The following file shows all the available settings for mTLS in a configuration resource:
+
+```
+apiVersion: dapr.io/v1alpha1
+kind: Configuration
+metadata:
+  name: default
+spec:
+  mtls:
+    enabled: true
+    workloadCertTTL: "24h"
+    allowedClockSkew: "15m"
+```
+
+The file here shows the default configuration settings. The examples below will show you how to change and apply this configuration to Sentry in Kubernetes and Self hosted modes.
+
+
+## Kubernetes
+
+### Setting up mTLS with the configuration resource
+
+In Kubernetes, Dapr creates a default configuration resource with mTLS enabled.
+
+Depending on how you install Dapr, this resource may reside in the `default` namespace if you installed Dapr using the Dapr CLI, or the `dapr-system` namespace if deployed using Helm.
+
+You can view the configuration resource with the following command:
+
+`kubectl get configurations/default --namespace <DAPR_NAMESPACE> -o yaml`.
+
+To make changes to the configuration resource, you can run the following command to edit it:
+
+```
+kubectl edit configurations/default --namespace <DAPR_NAMESPACE>
+```
+
+Once the changes are saved, you need to restart the Sentry pod in order for the changes to take effect.
+You can restart Sentry using the following command:
+
+```
+kubectl delete pod --selector=app=dapr-sentry --namespace <DAPR_NAMESPACE>
+```
+
+### Viewing logs
+
+In order to view Sentry logs, run the following command:
+
+```
+kubectl logs --selector=app=dapr-sentry --namespace <DAPR_NAMESPACE>
+```
+
+### Bringing your own certificates
+
+Using Helm, you can provide the PEM encoded root cert, issuer cert and private key that will be populated in the Kubernetes secret.
+
+*Note: This examples uses step to create the certificates. You can install step [here](https://smallstep.com/docs/getting-started/). Windows binaries available [here](https://github.com/smallstep/cli/releases)*
+
+Create the root certificate:
+
+```
+step certificate create cluster.local ca.crt ca.key --profile root-ca --no-password --insecure
+```
+
+Create the issuer certificate:
+
+```
+step certificate create cluster.local issuer.crt issuer.key --ca ca.crt --ca-key ca.key --profile intermediate-ca --not-after 8760h --no-password --insecure
+```
+
+This will create the root and issuer certs and keys.
+
+Install Helm and pass the root cert, issuer cert and issuer key to Sentry via configuration:
+
+```
+kubectl create ns dapr-system
+
+helm install \
+  --set-file dapr_sentry.tls.issuer.certPEM=issuer.crt \
+  --set-file dapr_sentry.tls.issuer.keyPEM=issuer.key \
+  --set-file dapr_sentry.tls.root.certPEM=ca.crt \
+  --namespace dapr-system \
+  dapr \
+  dapr/dapr
+```
+
+## Self-hosted
+
+### Running Sentry
+
+In order to run Sentry, you can either build from source, or download a release binary from [here](https://github.com/dapr/dapr/releases).
```

Comment:
Still do not get why we have to make the placement service easy and the sentry service harder to install locally. Seems like we need to revisit the CLI to help with this, if there are to be services that need to be configured first. This is just making more hoops to jump through

Reply:
Because the CLI is setting up a single machine environment, where mTLS will never run.

Second, you don't have to build from source: we have a binary ready to launch for every platform.